R/stacking.R

```diff
@@ -1142,7 +1142,13 @@ stackedDataSet <- function(input.data.set, input.data.set.metadata,
             input.var <- input.data.set[[i]]
             val.attr <- attr(input.var, "labels", exact = TRUE)
             if (isIntegerValued(input.var))
+            {
+                # DS-5236: Keep NaN as NaN not NA to avoid haven mangling excluded/missing value attributes on save
```

Comment:
- Does the same issue occur when combining data sets?
- I think input.var will become a non-integer vector if it has NaN as one of its elements. I don't think there is any way around it. Perhaps we could add an extra condition in the if: `!any(is.nan(input.var))` so that it is clear that input.var does not become integer valued in this situation.

Reply:
* Yes, appears the same change occurs with merging. will make a separate pr.
* i put in your suggested change to the `isIntegerValued` function and previous code